src/lib/route-builder.js

```diff
@@ -115,11 +109,23 @@ class RouteBuilder {
   * getDestinationTransfer (sourceTransfer) {
     log.info('constructing destination transfer ' +
       'sourceLedger=%s sourceAmount=%s ilpHeader=%s',
-      sourceTransfer.ledger, sourceTransfer.amount,
-      sourceTransfer.data && JSON.stringify(sourceTransfer.data.ilp_header))
-    const ilpHeader = sourceTransfer.data && sourceTransfer.data.ilp_header
-    if (!ilpHeader) {
-      throw new Error('source transfer is missing ilp_header in memo')
+      sourceTransfer.ledger, sourceTransfer.amount, sourceTransfer.ilp)
+    if (!sourceTransfer.ilp) {
+      throw new IlpError({
+        code: 'S01',
+        name: 'Invalid Packet',
+        message: 'source transfer is missing "ilp"'
+      })
+    }
+    let ilpHeader
+    try {
+      ilpHeader = packet.deserializeIlpPayment(Buffer.from(sourceTransfer.ilp, 'base64'))
+    } catch (err) {
+      throw new IlpError({
```

Comment:
If it's not going to report the specific error that was caught it should at least log it so we can find the exact problem when debugging